dbms/src/Common/SipHash.h

```diff
@@ -138,6 +139,11 @@ class SipHash
         update(reinterpret_cast<const char *>(&x), sizeof(x));
     }
 
+    void update(const std::string & x)
+    {
+        update(x.c_str(), x.length());
```

Comment:
data is more appropriate than c_str.
For example, in `fbstring` c_str method may lazily write trailing zero byte.